modules/geo-layers/src/tile-3d-layer/tile-3d-layer.js

```diff
@@ -249,18 +269,14 @@ export default class Tile3DLayer extends CompositeLayer {
         coordinateOrigin: cartographicOrigin,
         modelMatrix,
 
-        getColor: constantRGBA || this.props.color,
-        pickable: true,
-        numInstances: pointCount,
-        opacity: 1.0,
-        pointSize: 2
-      })
+        getColor: constantRGBA || getColor,
+        numInstances: pointCount
+      }
     );
   }
 
   renderLayers() {
-    const layerMapValues = Object.values(this.state.layerMap);
-    return layerMapValues.map(x => x.layer);
+    return Object.values(this.state.layerMap).map(layer => layer.layer);
```

Comment:
Shouldn't we precalculate this when we update the layerMap? Seems weird to do just one part of the calculation here.

Reply:
After each `updateState` will call `renderLayers`, so it is not quite necessary to precaculate the layers and cached in state in calling `_selectLayer` (part of `updateState`).

Reply:
Even so, to make this code simple to understand, we should keep all calculations in one place. It is confusing to have them in two places, as it looks like there is a reason for it, as if they are happening at different times and need to be kept separate for performance or other reasons.

Reply:
Why is _selectLayer needed at all? Can we just roll the logic into `renderLayers`?